core/src/main/java/com/linecorp/armeria/server/HttpServerHandler.java

```diff
@@ -197,16 +197,41 @@ public int unfinishedRequests() {
     }
 
     @Override
-    public void channelUnregistered(ChannelHandlerContext ctx) throws Exception {
-        super.channelUnregistered(ctx);
-        unfinishedResponses.keySet().forEach(StreamMessage::abort);
```

Comment:
No idea why `channelUnregistered` is used instead of `channelInactive` but they don't make any difference, so I moved the cleanup logic to `channelInactive`.

Reply:
The netty documentation on these callbacks wasn't very clear, so I implemented every method and added breakpoints to try to figure it out. This was called where I expected and unregistering from armeria when unregistering from netty made intuitive sense, but if they're similar semantics seems good.

Reply:
unregistration will always follow `channelInactive`. Netty allows deregistration without closing for a very special case, but we don't use it.